### PR TITLE
speed up syft unit tests

### PR DESCRIPTION
**Pull request by abyesilyurt**
## Description

I did some profiling on the unit tests and cleared up some bottlenecks.

There is roughly 35% speed up.

Local:
- before: 2 min.
- after: 1 min.

CI:
- before: 6 min.
- after: 4 min.

## Affected Dependencies
~~Added orjson library for faster json processing.~~ Hashing relies on the json file formatting and orjson is missing white spacing which is not configurable. So this change was reverted although there still some potential gains to be made.

## How has this been tested?
- unit tess

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: packages/syft/src/syft/service/user/user.py [hunk 1]
```diff
@@ -106,6 +108,7 @@ def salt_and_hash_password(password: str, rounds: int) -> tuple[str, str]:
     return salt_bytes, hashed_bytes
 
 
+@cache
```
Comment (eelcovdw) [inline packages/syft/src/syft/service/user/user.py]: Is the caching only to speed up CI? Not sure we want to do the caching with real passwords

Comment (abyesilyurt) [reply packages/syft/src/syft/service/user/user.py]: moved caching to conftest

File: packages/syft/tests/conftest.py [hunk 2]
```diff
@@ -204,3 +208,9 @@ def mongo_client(testrun_uid):
     "tests.syft.action_graph.fixtures",
     "tests.syft.serde.fixtures",
 ]
+# patching compute heavy hashing to speed up tests
+SyftClientSessionCache._get_key = (
+    lambda email, password, connection: f"{email}{password}{connection}"
+)
+user.salt_and_hash_password = cache(user.salt_and_hash_password)
+user.check_pwd = cache(user.check_pwd)
```
Comment (yashgorana) [inline packages/syft/tests/conftest.py]: This should be provisioned through `monkeypatch.setattr` in the fixture instead of making it a global patch?

Comment (abyesilyurt) [reply packages/syft/tests/conftest.py]: Good point! I applied the changes.

**Review by eelcovdw — APPROVED** [review]
Looks good!

**Pull request merged**